x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/common_model/ecs_event_builder.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EcsEventObjectKey, IEcsEvent } from './ecs_event';
+import { RuleExecutionEventLevel, getLevelSeverity } from './rule_execution_event_levels';
+import { RuleExecutionStatus, getStatusSeverity } from './rule_execution_statuses';
+
+const EVENT_LOG_PROVIDER = 'detection-engine'; // TODO: "siem", "siem-detection-engine", "security-solution", other?
+const EVENT_LOG_NAME = 'rule-execution-log'; // TODO: A more generic rule-log? A separate rule-management (rule-audit) log?
+
+export class EcsEventBuilder {
+  private _result: IEcsEvent = {};
+
+  constructor() {
+    // TODO: Which version does event_log use? Should it be specified here or inside the event log itself?
```

Comment:
The event log plugin provides this value, you don't need to provide it.  And probably shouldn't since we could end up with different versions in different documents.  Though I don't know of anything that makes use of this field right now.